invokeai/backend/generator/base.py

```diff
@@ -17,13 +22,215 @@
 from accelerate.utils import set_seed
 from diffusers import DiffusionPipeline
 from tqdm import trange
+from typing import List, Iterator
+from dataclasses import dataclass, field
+from diffusers.schedulers import SchedulerMixin as Scheduler
 
-import invokeai.assets.web as web_assets
 from ..util.util import rand_perlin_2d
+from ..safety_checker import SafetyChecker
+from ..prompting.conditioning import get_uc_and_c_and_ec
+from ..model_management.model_manager import ModelManager
+from ..stable_diffusion.diffusers_pipeline import StableDiffusionGeneratorPipeline
 
 downsampling = 8
-CAUTION_IMG = "caution.png"
 
+@dataclass
+class InvokeAIGeneratorBasicParams:
```

Comment:
So my issue with this is that it basically duplicates the node. Unless you were hoping to pull this out into a Pydantic class and then pass this around instead of having each node have the parameters?

Reply:
Doing this allows the `InvokeAIGenerator` class to be used outside of the nodes framework for quick one-off scripts.

In addition, there's going to be duplication somewhere if you want to pass arguments to a `generate()` method and have argument checking. If we don't have a `BasicParams` block, then we'd have a long call signature like this in the `generate()` method:

```
def generate(self, prompt: str, steps: int=20, perlin: float=0.0, noise_threshold: float=0.0,...)
```

If relying completely on the node for type checking, then the signature is `def generate(self,**kwargs)`, which doesn't feel right to me.